src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java

```diff
@@ -44,19 +62,63 @@ public void testJson() throws FileNotFoundException, URISyntaxException {
     Iterator<JsonValue> apis = reader.readArray().iterator();
     Assert.assertTrue(apis.hasNext());
     for (JsonObject api = (JsonObject) apis.next(); apis.hasNext(); api = (JsonObject) apis.next()) {
+      verifyApi(api);
+    }
+  }
+
+  private static void verifyApi(JsonObject api) throws URISyntaxException {
+    Assert.assertFalse(api.getString("name").isEmpty());
+    Assert.assertFalse(api.getString("description").isEmpty());
+    String transport = api.getString("transport");
+    Assert.assertTrue("http".equals(transport) || "grpc".equals(transport));
+    new URI(api.getString("documentation"));
+    new URI(api.getString("site"));
+    if (api.getString("icon") != null) {
+      new URI(api.getString("icon"));
+    }
+    JsonArray clients = api.getJsonArray("clients");
+    Assert.assertFalse(clients.isEmpty());
+    for (int i = 0; i < clients.size(); i++) {
+      JsonObject client = (JsonObject) clients.get(i);
+      String status = client.getString("status");
+      Assert.assertTrue(status, 
```

Comment:
Could use Hamcrest's `org.hamcrest.collection.IsArrayContaining` so the invalid value is shown in the failure message.

Reply:
done